T&A 45341/45342: Fixes import of solution values in single, multiple and kprim choice questions and vertical order questions

matheuszych opened the pull request:
https://mantis.ilias.de/view.php?id=45341
https://mantis.ilias.de/view.php?id=45342

These changes aim to fix the issues mentioned in the Mantis tickets.

matheuszych commented:
Already reviewed by @thojou.

kergomard commented:
Thank you very much @matheuszych !

I don't think this is really right, though:
- The database allows for 'value1' to be `null` so we cannot use `strcmp()` in `class.assMultipleChoiceGUI.php:306`.
- I think that the problem with `return $value1 ?: null;` is that it also captures '0', right? We should then change it to `$value1 !== '' ? $value1 : null`. Am I missing something?

Thank you very much again and best,
@kergomard

matheuszych committed:
Fixes import of solution values in single, multiple and kprim choice questions and vertical order questions

matheuszych commented:
Hello @kergomard 

I implemented your advice. 

The `strcmp` was used in `assSingleChoiceGUI` and `assMultipleChoiceGUI` a couple of times. I adjusted that. It is now checked whether `value1` is a string and only then, the `strcmp` is executed. `strcmp` does work here with strings and integers because of implicit type casting.

Sorry for the force push. The was not planned. 

Best reagrds
@matheuszych

kergomard commented on Modules/TestQuestionPool/classes/class.assMultipleChoiceGUI.php:
```diff
@@ -347,7 +347,7 @@ public function renderSolutionOutput(
                 if ($this->object->getSpecificFeedbackSetting() == 2) {
                     foreach ($user_solution as $mc_solution) {
-                        if (strcmp($mc_solution, $answer_id) == 0) {
+                        if (is_string($mc_solution) && strcmp($mc_solution, $answer_id) === 0) {
```
Sorry, I was not specific enough. This and all similar cases should simply be `$mc_solution === (string) $answer_id`. The `$answer_id` is an `int`.

matheuszych replied:
Changed.

kergomard reviewed (changes requested):
Thank you for the update @matheuszych ! I added a little comment for a change, see [here](https://github.com/ILIAS-eLearning/ILIAS/pull/9740/files#diff-d992b9232822011560c2de0377d4c8b986578814bd87ef318f6b6290976b6beeR350). Sorry for not being specific enough.

matheuszych committed:
Adds review adjustments

kergomard commented:
Thank you very much for bearing with me @matheuszych ! Merged and picked to ILIAS 10 and trunk.
